static/app/components/events/interfaces/frame/context.tsx

```diff
@@ -163,7 +187,12 @@ const Context = ({
 
   const lineColors: Array<Color | 'transparent'> =
     hasCoverageData && data!.codecov?.lineCoverage!
-      ? getCoverageColors(contextLines, data!.codecov?.lineCoverage)
+      ? getCoverageColors(
```

Comment:
just to be more clear, this function may get called any number of times during render

Reply:
you'll want to use a useEffect or useMemo to guarantee the number of times

Reply:
thank you! I used a `useMemo` for the function call. I also added `useMemo` for some of the other variables that are being computed - please let me know if the usage is correct

Reply:
@jangjodi I think the useRouteAnalyticsParams that @snigdhas recommended here makes more sense. You might return the information you want out of getCoverageColorClass or something.

The useMemo approach here is way too much for someone to reason about later just for analytics. You can simplify it next time by ignoring other dependencies since we probably only care about the data from the api and analytics don't need to be perfect.
```ts
useEffect(() => {
	if (data) {
		trackAnalyticsEvent
	}
// eslint-disable-next-line react-hooks/exhaustive-deps
}, [data])
```